pico/api/src/main/java/io/helidon/pico/api/DefaultQualifierAndValue.java

```diff
@@ -162,6 +162,26 @@ public static class Builder extends DefaultAnnotationAndValue.Builder {
         protected Builder() {
         }
 
+        @Override
+        public Builder typeName(TypeName val) {
+            return (Builder) super.typeName(val);
```

Comment:
This is the "hack" - as you do not have correctly generic parent, you cannot do this 100% right.
To avoid this cast, you can use
```
super.typeName(val);
return this;
```
to work around the explicit cast.

Reply:
pushed

Reply:
pushed